packages/bot-skeleton/src/scratch/hooks/workspace_svg.js

```diff
@@ -293,3 +294,90 @@ Blockly.WorkspaceSvg.prototype.paste = function(xml_block) {
         this.pasteBlock_(xml_block);
     }
 };
+
+/**
+ * Show the context menu for the workspace.
+ * @param {!Event} e Mouse event.
+ * @private
+ */
+Blockly.WorkspaceSvg.prototype.showContextMenu_ = function(e) {
+    if (this.options.readOnly || this.isFlyout) {
+        return;
+    }
+    const menu_options = [];
+    const top_blocks = this.getTopBlocks(true);
+    const all_blocks = this.getAllBlocks(true);
+    const event_group = Blockly.utils.genUid();
+    const ws = this;
+
+    // Options to undo/redo previous action.
+    menu_options.push(Blockly.ContextMenu.wsUndoOption(this));
+    menu_options.push(Blockly.ContextMenu.wsRedoOption(this));
+
+    // Option to clean up blocks.
+    if (this.scrollbar) {
+        menu_options.push(Blockly.ContextMenu.wsCleanupOption(this, top_blocks.length));
+    }
+
+    if (this.options.collapse) {
+        const has_collapsed_blocks = all_blocks.some(block => block.isCollapsed());
+        const has_expanded_blocks = all_blocks.some(block => !block.isCollapsed());
+
+        menu_options.push(Blockly.ContextMenu.wsExpandOption(has_collapsed_blocks, all_blocks));
+        menu_options.push(Blockly.ContextMenu.wsCollapseOption(has_expanded_blocks, all_blocks));
+    }
+
+    // Option to add a workspace comment.
+    if (this.options.comments) {
+        menu_options.push(Blockly.ContextMenu.workspaceCommentOption(this, e));
+    }
+
+    // Option to delete all blocks.
+    // Count the number of blocks that are deletable.
+    const delete_list = Blockly.WorkspaceSvg.buildDeleteList_(top_blocks);
+
+    // Scratch-specific: don't count shadow blocks in delete count
+    const delete_count = delete_list.filter(block => !block.isShadow()).length;
+
+    const DELAY = 10;
+    const deleteNext = () => {
+        Blockly.Events.setGroup(event_group);
+        const block = delete_list.shift();
+
+        if (block) {
+            if (block.workspace) {
+                block.dispose(false, true);
+                setTimeout(deleteNext, DELAY);
+            } else {
+                deleteNext();
+            }
+        }
+        Blockly.Events.setGroup(false);
+    };
+
+    const delete_option = {
+        text:
+            delete_count === 1
+                ? localize('Delete Block')
+                : localize('Delete {{ delete_count }} Blocks', { delete_count }),
+        enabled: delete_count > 0,
+        callback() {
+            if (ws.currentGesture_) {
+                ws.currentGesture_.cancel();
+            }
+            if (delete_count < 2) {
+                deleteNext();
+            } else {
+                const msg = localize('Delete all {{ delete_count }} blocks?', { delete_count });
+                Blockly.confirm(msg, function(ok) {
+                    if (ok) {
+                        deleteNext();
+                    }
+                });
+            }
+        },
+    };
```

Comment:
can we add this whole thing to `Blockly.contextMenu.wsDeleteOption`? just like others (wsExpandOption, wsCollapseOption)

Reply:
That makes sense 👍